### PR TITLE
feat(core): Create names for personal projects in one place and resuse it

### PR DESCRIPTION
**Pull request by despairblue**
## Summary

This adds a helper function to the user entity that generates the name for the user's personal project.
Next step would be to mark the field as non-nullable in the entity and the migration.

## Related tickets and issues

none

## Review / Merge checklist

- [x] PR title and summary are descriptive. **Remember, the title automatically goes into the changelog. Use `(no-changelog)` otherwise.** ([conventions](https://github.com/n8n-io/n8n/blob/master/.github/pull_request_title_conventions.md))
- [x] Tests included.
  > A bug is not considered fixed, unless a test is added to prevent it from happening again.
  > A feature is not complete without tests.

### CODE REVIEW

File: packages/cli/src/databases/entities/User.ts [hunk 1]
```diff
@@ -155,4 +155,14 @@ export class User extends WithTimestamps implements IUser {
 		const { password, apiKey, mfaSecret, mfaRecoveryCodes, ...rest } = this;
 		return rest;
 	}
+
+	createPersonalProjectName() {
+		if (this.firstName && this.lastName && this.email) {
+			return `${this.firstName} ${this.lastName} <${this.email}>`;
+		} else if (this.email) {
+			return `<${this.email}>`;
```
Comment (despairblue) [inline packages/cli/src/databases/entities/User.ts:163]: @krynble This is what it generates for shell accounts.

File: packages/cli/test/integration/users.api.test.ts [hunk 2]
```diff
@@ -543,7 +544,7 @@ describe('DELETE /users/:id', () => {
 	});
 
 	test('should fail to delete a user that does not exist', async () => {
-		await ownerAgent.delete('/users/foobar').query({ transferId: '' }).expect(404);
```
Comment (despairblue) [inline packages/cli/test/integration/users.api.test.ts:546]: Unrelated fix for the postgres tests.

File: packages/cli/src/databases/entities/User.ts (hunk 1 above)
Comment (despairblue) [reply packages/cli/src/databases/entities/User.ts:163]: Good idea.

They're here: 

https://github.com/n8n-io/n8n/pull/8995/files#diff-0157cebf7fcd16044b39d250c31aa3f3ae85e14d1afe27f4afb7dc17b0999225R21-R37

**Review by krynble — APPROVED** [review]
LGTM
